algebra/src/fields/models/quadratic_extension.rs

```diff
@@ -0,0 +1,495 @@
+use std::{
+    cmp::{Ord, Ordering, PartialOrd},
+    fmt,
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    io::{Read, Write, Result as IoResult},
+};
+use rand::{
+    distributions::{Distribution, Standard},
+    Rng,
+};
+
+use crate::{
+    bytes::{FromBytes, ToBytes},
+    bits::{FromBits, ToBits},
+    fields::{Field, FpParameters, LegendreSymbol, PrimeField, SquareRootField},
+    UniformRand, Error,
+};
+use crate::biginteger::arithmetic::find_wnaf;
+
+/// Model for quadratic extension field of prime field F=Fp
```

Comment:
Also here, this is the model for quadratic extensions of a BaseField, which is again an extension of BasePrimeField.